.cirrus.yml

```diff
@@ -574,7 +587,9 @@ task:
       script:
         - ulimit -S -n 2048 # https://github.com/flutter/flutter/issues/2976
         - dart --enable-asserts ./dev/bots/codesign.dart
-
+  cleanup_before_cache_upload_script:
+     # Remove volatile file to avoid unnecessary Cirrus cache invalidation.
+    - rm -f bin/cache/flutter_version_check.stamp
```

Comment:
Is there some way we can prevent this from breaking?  Maybe a test that checks that this is the only file changed?

Reply:
Why would this file invalidate the cirrus cache? Aren't we fingerprinting `bin/internal/*.stamp`?

Reply:
Ahhh.. do we use this in the tool to say we don't trust the cache?

Reply:
I know the shell wrappers don't check it. It looks to me like this is just used for checking for a new version and showing the user a warning.

Reply:
Other than making it readonly, I can't think how.  `bin/cache` is gitignored so there's no way to tell what's changed in the working directory.  Creating a sub-repo in that directory in setup to then reset before upload would take minutes to init and commit because of the large binaries.

https://github.com/flutter/flutter/issues/44526 would make this better.

Reply:
> Why would this file invalidate the cirrus cache? Aren't we fingerprinting `bin/internal/*.stamp`?

The cache invalidation doesn't come from the fingerprint, it comes from Cirrus's hashing of the cache.  From https://cirrus-ci.org/guide/writing-tasks/#cache-instruction
> After the last script instruction for the task succeeds, Cirrus CI will calculate checksum of the cached folder (note that it's unrelated to fingerprint_script instruction) and re-upload the cache if it finds any changes. To avoid a time-costly re-upload, remove volatile files from the cache (for example, in the last script instruction of a task).

Reply:
ahhh, i see. thus any changes of the cache during execution will trigger a re-upload.